### DISCUSSION TITLE
how to make filed required depending on relationship data

### DISCUSSION THREAD

**Discussion by Menks** [filamentphp/filament#11601] [Help] [answered]
### Package

Form builder

### Package Version

v3

### How can we help you?

Can someone please help me ?
Hello, I would like to make the copy_permis field required or not depending on the specialty selected

i have two models Candidat(id,name,specialty_id) and Specialty(id,name,is_required ). NB: is_required : is boolean

Forms\Components\Select::make('specialty_id')
                            ->required()
                            ->live()
                            ->searchable()->preload()
                            ->relationship('specialty','name'),

Forms\Components\FileUpload::make('copie_permis')
                            ->directory('permis_files')
                            ->openable()
                            ->downloadable()
                            ->image()
                            ->imageEditor()
                            ->required(fn (Get $get): bool => ($get('specialty_id.is_required') == true)),

**Comment by leandrocfe** [comment]
```php
->required(fn (Get $get): bool => Specialty::find($get('specialty_id'))->where('is_required', true)->exists())
```
?

**Reply by Menks** [reply]
Hi @leandrocfe thanks for your answer, I got the following error

Call to a member function where() on null

 My models

```
class Specialty extends Model
{

    use HasFactory;
    protected $fillable = ['name','is_required'];

    protected $casts = [
        'is_required' => 'boolean',
    ];

    public function candidats(): HasMany
    {
        return $this->hasMany(Candidat::class);
    }
}
```

```
class Candidat extends Model
{
    use HasFactory,SoftDeletes;
    protected $fillable = ['nom_complet', 'telephone', 'nni', 'copie_nni', 'date_naissance', 'categorie_permis', 'copie_permis', 'genre','nature_demande', 'observation', 'specialty_id', 'photo_file', 'certificat_med_deb_validite', 'certificat_med_fin_validite', 'copie_certificat_med' , 'casier_jud_deb_validite', 'casier_jud_fin_validite', 'copie_casier_jud', 'cv_file', 'status', 'motif_rejet', 'created_by', 'updated_by'];


    protected $casts =[
        'status' => CandidatStatus::class,
        'genre' => CandidatGenre::class,
        'nature_demande' => NatureDemande::class,
        'categorie_permis' => 'array',
        'telephone' => 'array',
    ];

    /**
     * @return BelongsTo
     */
    public function specialty(): BelongsTo
    {
        return $this->belongsTo(Specialty::class);
    }

    /**
     * @return BelongsTo
     */
    public function created_by(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function updated_by(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    /**
     * @return HasMany
     */
    public function diplomes(): HasMany
    {
        return $this->hasMany(DiplomeCandidat::class);
    }

    /**
     * @return HasMany
     */
    public function experiences(): HasMany
    {
        return $this->hasMany(ExperienceCandidat::class);
    }
}
```

**Comment by Menks** [comment, marked as answer]
->required(fn (Get $get): bool => Specialite::select('cp_requise')->where('id', $get('specialite_id'))->pluck('cp_requise')->join('/'))
